api/authenticator.py

```diff
@@ -1589,6 +1592,12 @@ def library(self, _db):
     def external_integration(self, _db):
         return get_one(_db, ExternalIntegration, id=self.external_integration_id)
 
+    @abstractmethod
+    def logging_name(self) -> str:
+        """The name used for this authentication provider in log messages."""
+        raise NotImplementedError()
+
+    @log_elapsed_time(log_method=log.info, message_prefix="authenticated_patron")
     def authenticated_patron(self, _db, header):
```

Comment:
Its slightly different from what we have done in other places, but maybe we could implement this as a `with` instead of a decorator. It would make naming the loggers a lot easier.

```python
@classmethod
def logger(cls):
    return logging.getLogger(f"{cls.__module__}.{cls.__name__}")

def authenticated_patron(self, _db, header):
    with elapsed_time_logging(
        log_method=self.logger.info
    ):
        ...
```

Then since the logger has the class name, we probably don't need to implement `logging_name` in each subclass.

Reply:
That looks good. I wasn't aware it was an option!

Reply:
We use different logger names all over the place, but I'd like to try to standardize on something like this, since its easy to implement, and it will let adjust the log level for different modules / classes if we need.

Reply:
Yep. The standard in Java land is to use the fully qualified class name. I don't think I've ever seen anything more than that be necessary, in practical terms.